opm/input/eclipse/EclipseState/IOConfig/FIPConfig.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+  Copyright 2015 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_FIP_CONFIG_HPP
+#define OPM_FIP_CONFIG_HPP
+
+#include <bitset>
+
+namespace Opm {
+
+class Deck;
+class DeckKeyword;
+class RPTConfig;
+
+//! \brief Class holding FIP configuration from RPTSOL/RPTSCHED keyword.
+class FIPConfig {
+public:
+    //! \brief Enumeration of FIP report outputs.
+    enum FIPOutputField {
```

Comment:
I know it's painful to use strong enums as indices, but I'd nevertheless really prefer that this was an `enum class` instead.  Mostly to avoid name collisions for common strings like `FIELD` and `FIP`.

Reply:
They have class scope so they are not FIELD and FIP but okay.

Reply:
Do you really insist? There are no potential of clashes here and the code turns complete turdy.

Reply:
Done. Not pretty.